src/services/services.ts

```diff
@@ -2072,7 +2079,8 @@ namespace ts {
             compilationSettings: CompilerOptions,
             scriptSnapshot: IScriptSnapshot,
             version: string,
-            acquiring: boolean): SourceFile {
+            acquiring: boolean,
+            scriptKind?: ScriptKind): SourceFile {
 
             const bucket = getBucketForCompilationSettings(compilationSettings, /*createIfMissing*/ true);

```

Comment:
How so? Isn't the bucket key based on the project compilation settings, whereas ScriptKind would be per file? I assume files still need to have unique paths (i.e. we couldn't have 2 files with the same path but different ScriptKinds).